src/sortStudents.ts

```diff
@@ -1,16 +1,69 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAvarageGrades(grades: number[]): number {
+  return [...grades].reduce((sum, el) => sum + el, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: string,
+): Student[] {
+  const copy: Student[] = [...students];
+  const result: Student[] = [];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return [...students].sort((a, b) => a.name.localeCompare(b.name));
+    case SortType.Surname:
+      return [...students].sort((a, b) => a.surname.localeCompare(b.surname));
```

Comment:
You can combine similar sort types, for example -> 
```suggestion
    case SortType.Name:
    case SortType.Surname:
      return [...students].sort((a, b) => a[sortBy].localeCompare(b[sortBy]));
```